### PR TITLE
Fixes Space Phase not making you space proof.

### PR DESCRIPTION
**Pull request by EnterTheJake**
## About The Pull Request

Space Phase now makes you space proof as it was intended.

## Why It's Good For The Game

Yes this is a fix, I had a convo with Comxia, the guy who designed Cosmic Heretic, and it turns out Space phase was intended to make you space proof.

image

It makes sense you literally phase out of existance, why would cold or pressure affect you?

## Changelog

:cl:
fix: Space Phase now makes the Heretic Space proof.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/heretic/magic/space_crawl.dm [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#define TRAIT_SPACE_PHASE "space-phasing"
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/magic/space_crawl.dm:1]:
```suggestion
#define SPACE_PHASING "space-phasing"
```

File: code/modules/antagonists/heretic/magic/space_crawl.dm [hunk 2]
```diff
@@ -82,6 +86,7 @@
 		jaunter.put_in_hands(left_hand)
 		jaunter.put_in_hands(right_hand)
 
+	jaunter.add_traits(jaunting_traits, TRAIT_SPACE_PHASE)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/magic/space_crawl.dm:89]:
```suggestion
	jaunter.add_traits(jaunting_traits, SPACE_PHASING)
```

File: code/modules/antagonists/heretic/magic/space_crawl.dm [hunk 3]
```diff
@@ -101,6 +106,7 @@
 
 	if(!exit_jaunt(jaunter, our_turf))
 		return FALSE
+	jaunter.remove_traits(jaunting_traits, TRAIT_SPACE_PHASE)
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/magic/space_crawl.dm:109]:
```suggestion
	jaunter.remove_traits(jaunting_traits, SPACE_PHASING)
```

File: code/modules/antagonists/heretic/magic/space_crawl.dm [hunk 4]
```diff
@@ -131,3 +137,5 @@
 /obj/item/space_crawl/Initialize(mapload)
 	. = ..()
 	ADD_TRAIT(src, TRAIT_NODROP, ABSTRACT_ITEM_TRAIT)
+
+#undef TRAIT_SPACE_PHASE
```
Comment (MrMelbert) [inline code/modules/antagonists/heretic/magic/space_crawl.dm:141]:
```suggestion
#undef SPACE_PHASING
```

**Review by MrMelbert — COMMENTED** [review]
Don't name trait sources, `TRAIT_[x]`. That's a bit misleading

Here's one example for how you could rename it

Optionally you can just do `REF(src)` or `type`. Both would also work

**Comment by uaioy** [comment]
the cosmic heretic buff is real

**Pull request merged**